microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java

```diff
@@ -0,0 +1,92 @@
+    private byte[] bytes;
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap
+                .create(WebArchive.class, FailSafeCDICustomConfigSourceHealthTest.class.getSimpleName() + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return webArchive;
+    }
+
+    @Before
+    public void backup() throws IOException {
+        bytes = Files.readAllBytes(propertyFile);
+    }
+
+    @After
+    public void restore() throws IOException {
+        Files.write(propertyFile, bytes);
+    }
+
+    @Override
+    protected void setConfigProperties(boolean live, boolean ready, boolean inMaintanance, boolean readyInMainenance)
+            throws Exception {
+        String content = String.format("%s=%s\n%s=%s\n%s=%s\n%s=%s",
+                FailSafeDummyService.LIVE_CONFIG_PROPERTY, live,
+                FailSafeDummyService.READY_CONFIG_PROPERTY, ready,
+                FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY, readyInMainenance,
+                FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY, inMaintanance);
+        // TODO Java 11 API way - Files.writeString(propertyFilePath, INCREMENT_CONFIG_PROPERTY + "=" + increment);
+        Files.write(propertyFile, content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Setup a microprofile-config-smallrye subsystem to obtain values from {@link CustomConfigSource}
+     */
+    static class SetupTask implements ServerSetupTask {
+        private static final String TEST_MODULE_NAME = "test.custom-config-source";
+
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
+            client.execute(String.format("/system-property=%s:add(value=%s)", CustomConfigSource.FILEPATH_PROPERTY,
+                    SetupTask.class.getResource(PROPERTY_FILENAME).getFile()));
+            ModuleUtil.add(TEST_MODULE_NAME)
+                    .setModuleXMLPath(SetupTask.class.getResource("configSourceModule.xml").getPath())
+                    .addResource("config-source", CustomConfigSource.class)
+                    .executeOn(client);
+            client.execute(String.format(
+                    "/subsystem=microprofile-config-smallrye/config-source=cs-from-class:add(class={module=%s, name=%s})",
+                    TEST_MODULE_NAME, CustomConfigSource.class.getName()));
+        }
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
```

Comment:
Please close client.

Reply:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

Reply:
Thanks - I understand limitations which you are running into but there is still an open client in the end...

Reply:
I see. There are two options:
- `ManagementClientProvider.onlineStandalone()` - this always opens a new connection which can be closed
- `ManagementClientProvider.onlineStandalone(managementClient)` - this just wraps `managementClient` and no other resources are opened thus it can be left opened. Since `managementClient` lifecycle is managed by Arquillian It must be left opened. And is closed by Arquillian at the end.

Reply:
Thanks - this issue was discussed off GitHub.

Reply:
And I know it is dangerous since Creaper wrapper can change in the future and open a resource which would be left in (2) option. However it is unlikely and @mnovak1 is ok with this (see https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/142)